<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>CorTiltWindow</class>
 <widget class="QMainWindow" name="CorTiltWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1578</width>
    <height>782</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Find COR and Tilt</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QVBoxLayout" name="verticalLayout_3">
    <item>
     <widget class="QSplitter" name="splitter">
      <property name="orientation">
       <enum>Qt::Horizontal</enum>
      </property>
      <widget class="QWidget" name="formLayoutWidget">
       <layout class="QVBoxLayout" name="verticalLayout" stretch="0,0">
        <property name="sizeConstraint">
         <enum>QLayout::SetMaximumSize</enum>
        </property>
        <property name="leftMargin">
         <number>9</number>
        </property>
        <property name="topMargin">
         <number>9</number>
        </property>
        <property name="rightMargin">
         <number>9</number>
        </property>
        <property name="bottomMargin">
         <number>9</number>
        </property>
        <item>
         <widget class="QTabWidget" name="tabWidget">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="MinimumExpanding">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="currentIndex">
           <number>0</number>
          </property>
          <widget class="QWidget" name="resultsTab">
           <attribute name="title">
            <string>Results</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout_4">
            <item>
             <widget class="QGroupBox" name="dataGroup">
              <property name="title">
               <string>Data</string>
              </property>
              <layout class="QFormLayout" name="formLayout">
               <item row="0" column="0" colspan="2">
                <widget class="StackSelectorWidgetView" name="stackSelector"/>
               </item>
               <item row="1" column="0" colspan="2">
                <widget class="QPushButton" name="setRoiBtn">
                 <property name="text">
                  <string>Set ROI from stack</string>
                 </property>
                </widget>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QGroupBox" name="resultGroup">
              <property name="maximumSize">
               <size>
                <width>16777215</width>
                <height>150</height>
               </size>
              </property>
              <property name="title">
               <string>Result</string>
              </property>
              <layout class="QHBoxLayout" name="horizontalLayout_2">
               <item>
                <layout class="QFormLayout" name="numericalResults">
                 <property name="fieldGrowthPolicy">
                  <enum>QFormLayout::AllNonFixedFieldsGrow</enum>
                 </property>
                 <item row="1" column="0">
                  <widget class="QLabel" name="resultCorLabel">
                   <property name="toolTip">
                    <string>Centre of Rotation, relative to full image coordinates.</string>
                   </property>
                   <property name="text">
                    <string>COR:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="1">
                  <widget class="QDoubleSpinBox" name="resultCor">
                   <property name="readOnly">
                    <bool>false</bool>
                   </property>
                   <property name="buttonSymbols">
                    <enum>QAbstractSpinBox::NoButtons</enum>
                   </property>
                   <property name="decimals">
                    <number>7</number>
                   </property>
                   <property name="maximum">
                    <double>99999.000000000000000</double>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="0">
                  <widget class="QLabel" name="resultTiltLabel">
                   <property name="toolTip">
                    <string>Tilt angle in degrees.</string>
                   </property>
                   <property name="text">
                    <string>Tilt:</string>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="1">
                  <widget class="QDoubleSpinBox" name="resultTilt">
                   <property name="readOnly">
                    <bool>false</bool>
                   </property>
                   <property name="buttonSymbols">
                    <enum>QAbstractSpinBox::NoButtons</enum>
                   </property>
                   <property name="prefix">
                    <string/>
                   </property>
                   <property name="suffix">
                    <string>Â°</string>
                   </property>
                   <property name="decimals">
                    <number>7</number>
                   </property>
                   <property name="minimum">
                    <double>-90.000000000000000</double>
                   </property>
                   <property name="maximum">
                    <double>90.000000000000000</double>
                   </property>
                  </widget>
                 </item>
                 <item row="3" column="1">
                  <widget class="QPushButton" name="pushButton">
                   <property name="toolTip">
                    <string>Use the CoR and Tilt above to generate a CoR for each of the slice indices in the table below</string>
                   </property>
                   <property name="toolTipDuration">
                    <number>2</number>
                   </property>
                   <property name="text">
                    <string>Calculate CoRs for slice indices</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
               <item>
                <layout class="QVBoxLayout" name="fitLayout"/>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <widget class="QGroupBox" name="manualGroup">
              <property name="title">
               <string>Manual</string>
              </property>
              <layout class="QVBoxLayout" name="verticalLayout_5">
               <item>
                <widget class="RemovableRowTableView" name="tableView"/>
               </item>
               <item>
                <layout class="QGridLayout" name="gridLayout">
                 <item row="0" column="0">
                  <widget class="QPushButton" name="removeBtn">
                   <property name="enabled">
                    <bool>false</bool>
                   </property>
                   <property name="toolTip">
                    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Removes the selected row from the table.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                   </property>
                   <property name="text">
                    <string>Remove</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="0">
                  <widget class="QPushButton" name="clearAllBtn">
                   <property name="toolTip">
                    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Removes all rows from the table.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                   </property>
                   <property name="text">
                    <string>Clear All</string>
                   </property>
                  </widget>
                 </item>
                 <item row="1" column="1">
                  <widget class="QPushButton" name="refineCorBtn">
                   <property name="enabled">
                    <bool>false</bool>
                   </property>
                   <property name="text">
                    <string>Refine</string>
                   </property>
                  </widget>
                 </item>
                 <item row="0" column="1">
                  <widget class="QPushButton" name="addBtn">
                   <property name="toolTip">
                    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Adds a new row to the table.&lt;/p&gt;&lt;p&gt;Slice index defaults to the current preview slice index.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                   </property>
                   <property name="text">
                    <string>Add</string>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="1">
                  <widget class="QPushButton" name="fitBtn">
                   <property name="toolTip">
                    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Performs linear fit of rows/points in the table.&lt;/p&gt;&lt;p&gt;Requires at least two rows.&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                   </property>
                   <property name="text">
                    <string>Fit</string>
                   </property>
                  </widget>
                 </item>
                 <item row="2" column="0">
                  <widget class="QPushButton" name="setAllButton">
                   <property name="enabled">
                    <bool>false</bool>
                   </property>
                   <property name="toolTip">
                    <string>&lt;html&gt;&lt;head/&gt;&lt;body&gt;&lt;p&gt;Sets the COR for all rows to match the COR of the selected row&lt;/p&gt;&lt;/body&gt;&lt;/html&gt;</string>
                   </property>
                   <property name="text">
                    <string>Set All</string>
                   </property>
                  </widget>
                 </item>
                </layout>
               </item>
              </layout>
             </widget>
            </item>
            <item>
             <spacer name="verticalSpacer_3">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>0</width>
                <height>0</height>
               </size>
              </property>
             </spacer>
            </item>
           </layout>
          </widget>
          <widget class="QWidget" name="reconTab">
           <attribute name="title">
            <string>Reconstruct</string>
           </attribute>
           <layout class="QVBoxLayout" name="verticalLayout_6">
            <item>
             <layout class="QGridLayout" name="optionsLayout">
              <item row="2" column="0">
               <widget class="QLabel" name="filterNameLabel">
                <property name="text">
                 <string>Reconstruction filter:</string>
                </property>
               </widget>
              </item>
              <item row="3" column="0">
               <widget class="QLabel" name="numIterLabel">
                <property name="text">
                 <string>Number of iterations:</string>
                </property>
               </widget>
              </item>
              <item row="3" column="1">
               <widget class="QSpinBox" name="numIter">
                <property name="minimum">
                 <number>1</number>
                </property>
                <property name="maximum">
                 <number>99999</number>
                </property>
               </widget>
              </item>
              <item row="1" column="0">
               <widget class="QLabel" name="algorithmNameLabel">
                <property name="text">
                 <string>Algorithm:</string>
                </property>
               </widget>
              </item>
              <item row="2" column="1">
               <widget class="QComboBox" name="filterName">
                <item>
                 <property name="text">
                  <string>shepp</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>cosine</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>hann</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>hamming</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>ramlak</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>parzen</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>butterworth</string>
                 </property>
                </item>
               </widget>
              </item>
              <item row="1" column="1">
               <widget class="QComboBox" name="algorithmName">
                <item>
                 <property name="text">
                  <string>gridrec</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>fbp</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>art</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>bart</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>mlem</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>osem</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>ospml_hybrid</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>ospml_quad</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>pml_hybrid</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>pml_quad</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>sirt</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>tv</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>grad</string>
                 </property>
                </item>
                <item>
                 <property name="text">
                  <string>tikh</string>
                 </property>
                </item>
               </widget>
              </item>
              <item row="0" column="0">
               <widget class="QLabel" name="maxProjAngleLabel">
                <property name="text">
                 <string>Maximum projection angle</string>
                </property>
               </widget>
              </item>
              <item row="0" column="1">
               <widget class="QDoubleSpinBox" name="maxProjAngle">
                <property name="maximum">
                 <double>9999.000000000000000</double>
                </property>
                <property name="value">
                 <double>360.000000000000000</double>
                </property>
               </widget>
              </item>
             </layout>
            </item>
            <item>
             <spacer name="verticalSpacer_5">
              <property name="orientation">
               <enum>Qt::Vertical</enum>
              </property>
              <property name="sizeHint" stdset="0">
               <size>
                <width>20</width>
                <height>40</height>
               </size>
              </property>
             </spacer>
            </item>
            <item>
             <layout class="QHBoxLayout" name="reconButtonGroup">
              <item>
               <widget class="QPushButton" name="reconstructSlice">
                <property name="text">
                 <string>Reconstruct Slice</string>
                </property>
               </widget>
              </item>
              <item>
               <widget class="QPushButton" name="reconstructVolume">
                <property name="text">
                 <string>Reconstruct Volume</string>
                </property>
               </widget>
              </item>
             </layout>
            </item>
           </layout>
          </widget>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="previewGroup">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Maximum" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="title">
           <string>Preview</string>
          </property>
          <layout class="QGridLayout" name="gridLayout_2">
           <item row="0" column="0">
            <widget class="QLabel" name="previewProjectionIndexLabel">
             <property name="text">
              <string>Projection index:</string>
             </property>
            </widget>
           </item>
           <item row="0" column="1">
            <widget class="QSpinBox" name="previewProjectionIndex"/>
           </item>
           <item row="1" column="1">
            <widget class="QSpinBox" name="previewSliceIndex">
             <property name="maximum">
              <number>9999</number>
             </property>
            </widget>
           </item>
           <item row="1" column="0">
            <widget class="QLabel" name="previewSliceIndexLabel">
             <property name="text">
              <string>Slice index:</string>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="verticalLayoutWidget">
       <layout class="QVBoxLayout" name="imageLayout">
        <property name="sizeConstraint">
         <enum>QLayout::SetMinimumSize</enum>
        </property>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <customwidgets>
  <customwidget>
   <class>StackSelectorWidgetView</class>
   <extends>QComboBox</extends>
   <header>mantidimaging.gui.widgets.stack_selector</header>
  </customwidget>
  <customwidget>
   <class>RemovableRowTableView</class>
   <extends>QTableView</extends>
   <header>mantidimaging.gui.widgets</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
